docs/servo_power_monitor/src/index.ts

```diff
@@ -166,47 +198,63 @@ function setupStartUSBButton() {
 };
 setupStartUSBButton();
 
-requestSerialButton.addEventListener('click', () => {
+requestSerialButton.addEventListener('click', async () => {
   halt = false;
-  navigator.serial
-      .requestPort({filters: [{usbVendorId: 0x18d1, usbProductId: 0x520d}]})
-      .then(async (port) => {
-        // Connect to `port` or add it to the list of available ports.
-        await port.open({baudRate: 115200});
-        const encoder = new TextEncoder();
-        const writer = port.writable.getWriter();
-        await writer.write(encoder.encode('help\n'));
-        writer.releaseLock();
-
-        kickWriteLoop(async (s) => {
-          let data = new TextEncoder().encode(s);
-          const writer = port.writable.getWriter();
-          await writer.write(data);
-          writer.releaseLock();
-        })
-        readLoop(async () => {
-          const reader = port.readable.getReader();
-          try {
-            while (true) {
-              const {value, done} = await reader.read();
-              if (done) {
-                // |reader| has been canceled.
-                break;
-              }
-              return utf8decoder.decode(value);
-            }
-          } catch (error) {
-            console.error(error);
-            throw error;
-          } finally {
-            reader.releaseLock();
-          }
-        });
-      })
-      .catch((e) => {
-        // The user didn't select a port.
-        console.error(e);
-      });
+  port = await navigator.serial
+  .requestPort({filters: [{usbVendorId: 0x18d1, usbProductId: 0x520d}]})
+  .catch((e) => {
+    console.error(e);
+  });
+  await port.open({baudRate: 115200});
+  requestSerialButton.disabled = true;
+  const encoder = new TextEncoder();
+  const writer = port.writable.getWriter();
+  await writer.write(encoder.encode('help\n'));
+  writer.releaseLock();
+
+  kickWriteLoop(async (s) => {
+    let data = new TextEncoder().encode(s);
+    const writer = port.writable.getWriter();
+    await writer.write(data);
+    writer.releaseLock();
+  })
+  readLoop(async () => {
+    reader = port.readable.getReader();
+    try {
+      while (true) {
+        const {value, done} = await reader.read();
+        if (done) {
+          // |reader| has been canceled.
+          break;
+        }
+        return utf8decoder.decode(value);
+      }
+    } catch (error) {
+      console.error(error);
+      throw error;
+    } finally {
+      reader.releaseLock();
+    }
+  });
+});
+
+// event when you disconnect USB port
+navigator.usb.addEventListener("disconnect", () => {
+  if (requestUSBButton.disabled) {
+    halt = true;
+    requestUSBButton.disabled = false;
+    inProgress = false;
+    // USB port is closed by specification when device is disconnected
```

Comment:
nit: I would say:
```
//  No need to call close() for the USB port here because the specification says that
// the port will be closed automatically when a device is disconnected.
// c.f. <paste the URL to a section in the spec which says so>
```

Reply:
maybe the nearest URL is https://wicg.github.io/webusb/#disconnect, so I removed the last sentence.